### PR TITLE
grt: add command to route specific nets

### PR DESCRIPTION
**Pull request by eder-matheus**
Fixes https://github.com/The-OpenROAD-Project/OpenROAD/issues/1747

### CODE REVIEW

File: src/grt/src/GlobalRouter.tcl [hunk 1]
```diff
@@ -340,6 +340,26 @@ proc repair_antennas { args } {
...
+                 flags {}
+  sta::check_argc_eq1 "set_nets_to_route" $args
+  set net_names [lindex $args 0]
+  set block [ord::get_db_block]
+  if { $block == "NULL" } {
+    utl::error GRT 252 "Missing dbBlock."
+  }
+
+  foreach net [get_nets $net_names] {
+    if { $net != "NULL" } {
```
Comment (maliberty) [inline src/grt/src/GlobalRouter.tcl:357]: warn if not found

File: src/grt/src/GlobalRouter.cpp [hunk 2]
```diff
@@ -2743,35 +2743,47 @@ static bool nameLess(const Net* a, const Net* b)
...
+    std::sort(non_clk_nets.begin(), non_clk_nets.end(), nameLess);
+
+    nets = clk_nets;
+    nets.insert(nets.end(), non_clk_nets.begin(), non_clk_nets.end());
+  } else {
+    for (odb::dbNet* db_net : nets_to_route_) {
+      Net* net = addNet(db_net);
+      if (net) {
+        nets.push_back(net);
+      }
```
Comment (maliberty) [inline src/grt/src/GlobalRouter.cpp:2783]: shouldn't we still sort these by clk/non-clk?

**Review by maliberty — CHANGES_REQUESTED** [review]

File: src/grt/src/GlobalRouter.tcl (hunk 1 above)
Comment (eder-matheus) [reply src/grt/src/GlobalRouter.tcl:357]: The `get_nets` function already has a warning for nets not found.

File: src/grt/src/GlobalRouter.cpp (hunk 2 above)
Comment (eder-matheus) [reply src/grt/src/GlobalRouter.cpp]: Fixed it, and the code is better now, with fewer changes than before.

**Review by maliberty — APPROVED** [review]

**Pull request merged**